### PR TITLE
fix: insert tolerance into `redeemShare()` checks [SUP-8853]

### PR DESCRIPTION
**Pull request by TamaraRingas**
https://github.com/yAudit/superform-report/issues/2

deposit4626() is used to redeem vault assets and deposit them in a superform vault. However, in _redeemShare(), there is a strict equality balance check, that, in case of stETH, will always revert.

Technical Details

Lido stETH has a know rounding down problem as stated in their [docs](https://docs.lido.fi/guides/lido-tokens-integration-guide/#1-2-wei-corner-case). When trying to redeem stETH with [deposit4626()](https://github.com/superform-xyz/superform-core/blob/f7f06d763a6af148899e9c292f1a57414db60cc3/src/router-plus/SuperformRouterPlus.sol#L356-L390), the [balance check](https://github.com/superform-xyz/superform-core/blob/f7f06d763a6af148899e9c292f1a57414db60cc3/src/router-plus/SuperformRouterPlus.sol#L522) will always fail leading the function to revert.

Impact

Low. It is not possible to use deposit4626() with a stETH vault and any other token that have a similar behaviour.

Recommendation

Avoid to use strict equality and insert a tolerance in the balance check to handle the 1-2 wei corner case in the same way that is done [here](https://github.com/superform-xyz/superform-core/blob/f7f06d763a6af148899e9c292f1a57414db60cc3/src/forms/ERC5115Form.sol#L30-L31).

### CODE REVIEW

File: src/router-plus/SuperformRouterPlus.sol [hunk 1]
```diff
@@ -521,6 +524,7 @@ contract SuperformRouterPlus is ISuperformRouterPlus, BaseSuperformRouterPlus {
         if (
             (balanceDifference != assets)
                 || (ENTIRE_SLIPPAGE * assets < ((expectedOutputAmount_ * (ENTIRE_SLIPPAGE - maxSlippage_))))
+                || (assets - TOLERANCE_CONSTANT > balanceDifference)
```
Comment (0xTimepunk) [inline src/router-plus/SuperformRouterPlus.sol:527]: Think this if statement is really big. Is it possible to use a new error (TOLERANCE_EXCEEDED)?

File: src/router-plus/SuperformRouterPlus.sol [hunk 2]
```diff
@@ -27,6 +27,9 @@ contract SuperformRouterPlus is ISuperformRouterPlus, BaseSuperformRouterPlus {
 
     uint256 public ROUTER_PLUS_PAYLOAD_ID;
 
+    /// @dev Tolerance constant to account for tokens with rounding issues on transfer
+    uint256 constant TOLERANCE_CONSTANT = 2 wei;
```
Comment (0xTimepunk) [inline src/router-plus/SuperformRouterPlus.sol:31]: Let's use 10 WEI just like in 5115

**Review by 0xTimepunk — COMMENTED** [review]
Almost there!

File: src/router-plus/SuperformRouterPlus.sol (hunk 1 above)
Comment (TamaraRingas) [reply src/router-plus/SuperformRouterPlus.sol:527]: Sure thing!

File: src/router-plus/SuperformRouterPlus.sol (hunk 2 above)
Comment (TamaraRingas) [reply src/router-plus/SuperformRouterPlus.sol:31]: Okay cool 👍

**Review by 0xTimepunk — APPROVED** [review]

**Pull request merged**